src/services/book-service.ts

```diff
@@ -0,0 +1,115 @@
+import { Request } from 'express';
+import { Document, Query, Schema, SortOrder } from 'mongoose';
+
+import { Book, IBook } from '../models/book';
+import { IBookQuery } from '../interfaces/book-query';
+import { UserRoleEnum } from '../utils/common/enum';
+import {
+  COMMON_QUERIES,
+  BOOK_NON_STRING_QUERIES
+} from '../utils/common/book-filters';
+
+import { toLowerCaseQuery } from '../utils/functions/lower-case-query';
+
+class BookService {
+  public async editSpecificBook(
+    bookId: string,
+    userId: Schema.Types.ObjectId | undefined,
+    updatedBook: IBook
+  ): Promise<Document<any, any, IBook> | null> {
+    return Book.findOneAndUpdate(
+      {
+        _id: bookId,
+        user_id: userId
+      },
+      updatedBook,
+      { new: true }
+    );
+  }
+
+  public async deleteSpecificBook(
+    bookId: string,
+    userId: Schema.Types.ObjectId | undefined
+  ): Promise<Document<any, any, IBook> | null> {
+    return Book.findOneAndDelete({
+      _id: bookId,
+      user_id: userId
+    });
+  }
+
+  public async getBookList(
+    userId: Schema.Types.ObjectId | undefined,
+    filters: IBookQuery,
+    limit: string,
+    skip: string,
+    sort: { [key: string]: SortOrder } | undefined,
+    userRole: string
+  ): Promise<Query<any, Document<IBook>>> {
+    let query = {};
+    switch (userRole) {
+      case UserRoleEnum.USER:
+        query = {
+          $or: [{ user_id: userId }, { enabled: true }],
+          ...filters
+        };
+        break;
+      case UserRoleEnum.ADMIN:
+        query = {
+          ...filters
+        };
+        break;
+      default:
+        query = {
+          ...filters,
+          enabled: filters.enabled === false ? null : true
+        };
+    }
+    return Book.find(query, null).sort(sort).limit(+limit).skip(+skip);
```

Comment:
Hạn chế dùng những hack cheat thế này. Dùng parseInt sẽ tốt hơn mặc dù dài dòng

```suggestion
    return Book.find(query, null)
                     .sort(sort)
                     .limit(parseInt(limit) || 0)
                     .skip(parseInt(skip) || 0);
```